File: tests_wraps/src/main/java/com/github/klee0kai/test/boxed/model/CarBoxed.java
package com.github.klee0kai.test.boxed.model;

import com.github.klee0kai.test.car.model.Bumper;
import com.github.klee0kai.test.car.model.Wheel;
import com.github.klee0kai.test.car.model.Window;

import java.util.Collections;
import java.util.List;
import java.util.UUID;

public class CarBoxed {

    public static int createCount = 0;
    public String uuid = UUID.randomUUID().toString();

    public List<CarBox<Bumper>> bumpers;
    public List<CarBox<Wheel>> wheels;
    public List<CarBox<Window>> windows;

    public CarBoxed(
            CarBox<Bumper> bumper,
            CarBox<Wheel> wheel,
            CarBox<Window> window
    ) {
        createCount++;
        bumpers = Collections.singletonList(bumper);
        wheels = Collections.singletonList(wheel);
        windows = Collections.singletonList(window);
    }


    public CarBoxed(List<CarBox<Bumper>> bumpers, List<CarBox<Wheel>> wheels, List<CarBox<Window>> windows) {
        createCount++;
        this.bumpers = bumpers;
        this.wheels = wheels;
        this.windows = windows;
    }
}

File: tests_wraps_kotlin/src/main/kotlin/com/github/klee0kai/test/boxed/model/CarBoxed.kt
package com.github.klee0kai.test.boxed.model

import com.github.klee0kai.test.car.model.Bumper
import com.github.klee0kai.test.car.model.Wheel
import com.github.klee0kai.test.car.model.Window
import java.util.*

class CarBoxed {
    var uuid = UUID.randomUUID().toString()
    var bumpers: List<CarBox<Bumper>>
    var wheels: List<CarBox<Wheel>>
    var windows: List<CarBox<Window>>

    constructor(
        bumper: CarBox<Bumper>,
        wheel: CarBox<Wheel>,
        window: CarBox<Window>
    ) {
        createCount++
        bumpers = listOf(bumper)
        wheels = listOf(wheel)
        windows = listOf(window)
    }

    constructor(bumpers: List<CarBox<Bumper>>, wheels: List<CarBox<Wheel>>, windows: List<CarBox<Window>>) {
        createCount++
        this.bumpers = bumpers
        this.wheels = wheels
        this.windows = windows
    }

    companion object {
        var createCount = 0
    }
}
